surreal/client.py

```diff
@@ -0,0 +1,75 @@
+"""
+Copyright © SurrealDB Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from typing import Any
+
+from surreal.http import HTTPClient
+
+
+class Surreal:
+    """
+    This class wraps around the HTTP Client adding a few 
+    fundemental things like list removal on single lengthed execution returns.
+
+    Parameters
+    ----------
+    url: :class:`str`
+        The url your surrealdb instance is running on.
+    username: :class:`str`
+        The username to use to authenticate with your instance.
+    password: :class:`str`
+        The password to use to authenticate with your instance.
+    ns: :class:`str`
+        The namespace on your instance to use.
+    db: :class:`str`
+        The database on your instance to use.
+
+    Properties
+    ----------
+    http: :class:`HTTPClient`
+    """
+    def __init__(
+        self, url: str, username: str, password: str, ns: str, db: str
+    ) -> None:
+        self.http = HTTPClient(
+            url=url, username=username, password=password, ns=ns, db=db
+        )
+
+    async def execute(self, query: str) -> dict[str, Any] | list[dict[str, Any]]:
+        """
+        Executes this query and returns the data given by the server.
+
+        If the query is a transactional query, the list should be ordered based on the transaction order.
+
+        Parameters
+        ----------
+        query: :class:`str`
+            The query to execute.
+
+        Returns
+        -------
+        row: dict[str, Any]
+        rows: list[dict[str, Any]]
+        """
+
+        execution = await self.http.execute(query=query)
+
+        return execution[0] if len(execution) == 1 else execution
```

Comment:
I can't speak for everyone here but I think the extra "smart-ness" employed here to return either a single dict or a list of dicts is trying to do too much. I'd like to hear others' thoughts on this, but it feels like too much black box magic is trying to be done, which results in more confusing logic for the end-user, *especially* if they're using typing, which makes this return type complex to deal with.

Reply:
I agree, better to be consistent with returns. The user would have to check for type of return otherwise

Reply:
^, there should be a separate method for each.